src/starbase/setup.ts

```diff
@@ -47,10 +47,16 @@ async function checkInstanceConfigFields(
         // of our integrations.
         const capsConfig = Object.assign(
           {},
-          ...Object.keys(invocationConfig.instanceConfigFields).map((key) => ({
-            [snakecase(key).toUpperCase()]:
-              invocationConfig.instanceConfigFields[key],
-          })),
+          ...Object.keys(invocationConfig.instanceConfigFields)
```

Comment:
Is this something that we can add a test for?

Also, I think this would be easier to read if we split it into a few functions with clear names. Here are a few suggestions to consider (untested):

```ts
function getRequiredIntegrationConfigFields(
  instanceConfigFields: IntegrationInstanceConfigFieldMap,
) {
  return Object.keys(instanceConfigFields).filter(
    (key) => !!instanceConfigFields[key]?.optional,
  );
}

interface IntegrationInvocationConfigWithConfigFields
  extends IntegrationInvocationConfig {
  instanceConfigFields: IntegrationInstanceConfigFieldMap<IntegrationInstanceConfig>;
}

/**
 * Converts the invocation config into the environment variable schema format
 * that integrations validate during execution.
 *
 * For example:
 *
 * `apiKey` => `API_KEY`
 */
function invocationConfigToEnvFormat(
  invocationConfig: IntegrationInvocationConfigWithConfigFields,
) {
  return getRequiredIntegrationConfigFields(
    invocationConfig.instanceConfigFields,
  ).map((key) => ({
    [upperSnake(key)]: invocationConfig.instanceConfigFields[key],
  }));
}

function upperSnake(s: string) {
  return snakecase(s).toUpperCase();
}

/**
 * After we have cloned the repository, pull down information on the required
 * instanceConfigFields and perform additional checks.  Returns false if the
 * config fields weren't successfully validated.
 */
async function checkInstanceConfigFields(
  integration: StarbaseIntegration,
): Promise<boolean> {
  let checkSuccessful = true;

  const { invocationConfig } = await import(
    `../.${integration.directory}/src/index`
  );

  if (!invocationConfig.instanceConfigFields) return checkSuccessful;

  // We have to snake_case and UPPERCASE our instanceConfigFields keys before
  // building our validator to mimic conversions that occur during execution
  // of our integrations.
  const capsConfig = Object.assign(
    {},
    ...invocationConfigToEnvFormat(invocationConfig),
  );

  const integrationSchema: Schema = {
    type: 'object',
    properties: capsConfig,
    required: Object.keys(capsConfig),
  };

  const validator = ajv.compile(integrationSchema);

  if (!validator(integration.config)) {
    // Log but don't throw an error so we can report errors for all integration configurations, not just the first failure
    console.error(
      `ERROR:  instanceConfigFields validation error(s) for ${integration.name}:  `,
      validator.errors,
    );

    checkSuccessful = false;
  }

  return checkSuccessful;
}
```